### PR TITLE
Introduce the new faceting and pagination settings

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR introduces two new settings following the newly created spec https://github.com/meilisearch/specifications/pull/157:
 - The `faceting.max_values_per_facet` one describes the maximum number of values (each with a count) associated with a value in a facet distribution query.
 - The `pagination.limited_to` one describes the maximum number of documents that a search query can ever return.

### CODE REVIEW

File: meilisearch-http/src/routes/indexes/settings.rs [hunk 1]
```diff
@@ -282,6 +282,50 @@ make_setting_route!(
...
+    "faceting",
+    analytics,
+    |setting: &Option<meilisearch_lib::index::updates::FacetingSettings>, req: &HttpRequest| {
+        use serde_json::json;
+
+        analytics.publish(
+            "Faceting Updated".to_string(),
+            json!({
+                "faceting": {
+                    "max_values_per_facet": setting.as_ref().and_then(|s| s.max_values_per_facet.set()),
```
Comment (Kerollmops) [inline meilisearch-http/src/routes/indexes/settings.rs:299]: @irevoire Could you check this line, is it correct [according to the specs](https://github.com/meilisearch/specifications/pull/157/files#diff-dbac052211a8ea4b2c5d068a6264380740b022efead552b4dfc9e4e8d961f0f4)?

File: meilisearch-http/src/routes/indexes/settings.rs [hunk 2]
```diff
@@ -282,6 +282,50 @@ make_setting_route!(
...
+);
+
+make_setting_route!(
+    "/pagination",
+    patch,
+    meilisearch_lib::index::updates::PaginationSettings,
+    pagination,
+    "pagination",
+    analytics,
+    |setting: &Option<meilisearch_lib::index::updates::PaginationSettings>, req: &HttpRequest| {
```
Comment (Kerollmops) [inline meilisearch-http/src/routes/indexes/settings.rs:314]: @irevoire Could you check this line, is it correct [according to the specs](https://github.com/meilisearch/specifications/pull/157/files#diff-dbac052211a8ea4b2c5d068a6264380740b022efead552b4dfc9e4e8d961f0f4)?

File: meilisearch-http/src/routes/indexes/settings.rs (hunk 1 above)
Comment (irevoire) [reply meilisearch-http/src/routes/indexes/settings.rs:299]: yep it looks good to me :+1:

File: meilisearch-http/src/routes/indexes/settings.rs (hunk 2 above)
Comment (irevoire) [reply meilisearch-http/src/routes/indexes/settings.rs:314]: perfecto :ok_hand:

**Review by irevoire — APPROVED** [review]
it looks perfect to me :+1:

**Pull request merged**